### ISSUE TITLE
RUF059: unused-unpacked-variable not working in v0.13.3

### ISSUE THREAD

**Issue by ayorgo** [astral-sh/ruff#24342]
### Summary

Unless my setup is somehow botched up, the rule doesn't seem to be working on the  [advertised](https://docs.astral.sh/ruff/rules/unused-unpacked-variable/) example.

Steps to reproduce:
```sh
$ ruff --version
ruff 0.13.3

$ cat <<EOF > unused-unpacked-variable.py
def get_pair():
    return 1, 2


def foo():
    x, y = get_pair()
    return x
EOF

$ cat unused-unpacked-variable.py        
def get_pair():
    return 1, 2


def foo():
    x, y = get_pair()
    return x

$ ruff check unused-unpacked-variable.py 
All checks passed!

$ ruff check --isolated unused-unpacked-variable.py 
All checks passed!
```

`pyproject.toml`
```toml
[tool.ruff]
line-length = 88
indent-width = 4
include = ["*.py", "**/pyproject.toml"]

[tool.ruff.lint]
ignore = ["E712"]
select = [ "E4", "E7", "E9", "B", "SIM", "I", "F", ]
```

### Version

0.13.3

**Comment by MichaReiser** [comment]
It seems your configuration doesn't select `RUF059`, which is probably the reason why it doesn't flag your code.

**Comment by ayorgo** [comment]
Hmm, indeed, thank you @MichaReiser. Had to enable the preview mode for it to work:

`pyproject.toml`:
```toml
preview = true
ignore = ["E712"]
select = ["E4", "E7", "E9", "B", "SIM", "I", "F", "RUF059"]
```

Could be a documentation issue then as it's not clear from the docs that this feature is in preview mode and needs to be explicitly `select`ed to work.

**Comment by MichaReiser** [comment]
@ayorgo It shouldn't require preview. The rule was stabilized in 0.13.0

```
❯ uvx ruff@0.13.0 check unpack.py --isolated --select RUF059
RUF059 Unpacked variable `y` is never used
 --> unpack.py:6:8
  |
5 | def foo():
6 |     x, y = get_pair()
  |        ^
7 |     return x
  |
help: Prefix it with an underscore or any other dummy variable pattern

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

**Comment by ayorgo** [comment]
Ah, my bad. Reverted my code to 0.9.10 accidentally.

Why is the rule not a default then? Seems like a natural extension of [unused-variable (F841)](https://docs.astral.sh/ruff/rules/unused-variable/#unused-variable-f841) which I thought covered cases like above until I was corrected in a code review about an unused variable.

Is it due to it not being implemented by Flakes historically?

On the other hand, I understand that putting all the rules from all the different tools under one "ruff" is very hard so maybe I should just do more of RTFM.

Thank you for your time @MichaReiser!
